cli/commands/catalog/action.go

```diff
@@ -39,7 +40,10 @@ func Run(ctx context.Context, opts *options.TerragruntOptions, repoURL string) e
 	for _, repoURL := range repoURLs {
 		tempDir := filepath.Join(os.TempDir(), fmt.Sprintf(tempDirFormat, util.EncodeBase64Sha1(repoURL)))
 
-		repo, err := module.NewRepo(ctx, opts.Logger, repoURL, tempDir)
+		experiment := opts.Experiments[experiment.Symlinks]
+		walkWithSymlinks := experiment.Evaluate(opts.ExperimentMode)
```

Comment:
Since this code is the same for all `repoURLs`, it might be better to move it out of the `for` block.